mlflow/transformers.py

```diff
@@ -734,7 +738,7 @@ def log_model(
     """
     return Model.log(
         artifact_path=artifact_path,
-        flavor=mlflow.transformers,
+        flavor=sys.modules[__name__],  # Get the current module.
```

Comment:
I don't get it, what issue is fixed by this line ?

btw, other flavors such as sklearn also uses `flavor=mlflow.sklearn` in model logging.

Reply:
It's a good practice to avoid module-level import and use direct imports inside the package itself. In this file we are doing both 
```
import mlflow
```
and direct imports like `from mlflow import xxx`.

By using `sys.modules[__name__]` we can fully switch to direct imports.

Reply:
in fact `mlflow.transformer` it is not a "import" command, we have `from mlflow import transformers` in `mlflow/__init__.py`

Reply:
for `import mlflow`, we uses it in many sub modules, if we want to modify it, we should modify all in one PR. it might be a lot of work

Reply:
> we should modify all in one PR

why?

Reply:
This isn't a formatting issue.

Reply:
+1 prefer progressive PRs instead of forcing one mega one, which would also be hard for making rollbacks. 

@WeichenXu123 Re `mlflow.transformer`, function-wise it's completely fine. The thing we want to avoid is `import mlflow` line inside our package.